src/compute.ts

```diff
@@ -13,50 +13,79 @@
 // limitations under the License.
 
 import * as aws from "@pulumi/aws";
+
 import { EnforcementLevel, ResourceValidationPolicy, validateTypedResource } from "@pulumi/policy";
 
-export const compute: ResourceValidationPolicy[] = [
-    ec2InstanceDetailedMonitoringEnabled("mandatory"),
-    ec2InstanceNoPublicIP("mandatory"),
-    ec2VolumeInUseCheck("mandatory", true),
-    elbAccessLoggingEnabled("mandatory"),
-    encryptedVolumes("mandatory"),
-];
+import { registerPolicy } from "./awsGuard";
+import { defaultEnforcementLevel } from "./enforcementLevel";
+import { PolicyArgs } from "./policyArgs";
+import { getValueOrDefault } from "./util";
+
+// Mixin additional properties onto AwsGuardArgs.
+declare module "./awsGuard" {
+    interface AwsGuardArgs {
+        ec2InstanceDetailedMonitoringEnabled?: EnforcementLevel;
+        ec2InstanceNoPublicIP?: EnforcementLevel;
+        ec2VolumeInUseCheck?: EnforcementLevel | Ec2VolumeInUseCheckArgs;
+        elbAccessLoggingEnabled?: EnforcementLevel;
+        encryptedVolumes?: EnforcementLevel | EncryptedVolumesArgs;
+    }
+}
+
+// Register policy factories.
+registerPolicy("ec2InstanceDetailedMonitoringEnabled", ec2InstanceDetailedMonitoringEnabled);
+registerPolicy("ec2InstanceNoPublicIP", ec2InstanceNoPublicIP);
+registerPolicy("ec2VolumeInUseCheck", ec2VolumeInUseCheck);
+registerPolicy("elbAccessLoggingEnabled", elbAccessLoggingEnabled);
+registerPolicy("encryptedVolumes", encryptedVolumes);
```

Comment:
Register the factories. This looks like magic strings that could be misspelled, but we actually have proper type checking here.

Reply:
Hopefully there is "a feature for that"(tm)
https://twitter.com/aChrisSmith/status/1199096162101317633

Reply:
Would love it if there was.

We could get away with just passing the functions themselves, giving up some compiler checking for runtime checking:

```typescript
registerPolicy(ec2InstanceDetailedMonitoringEnabled);
registerPolicy(ec2InstanceNoPublicIP);
registerPolicy(ec2VolumeInUseCheck);
// etc.
```

This assumes the functions have the same name as the property in `AwsGuardArgs`, and they're declared using `function ec2InstanceDetailedMonitoringEnabled(...)` (which they are in our case), which allows us to use the function's [`name`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) property to get the string. But we'd be giving up some compile time safety. That's probably fine honestly. The runtime checks would fail during testing, if there was a problem.

Reply:
Left this as-is for now.